### DISCUSSION TITLE
2.6.0

### DISCUSSION THREAD

**Discussion by caring-coder** [verronpro/office-stamper#509] [Announcements]
## Features

* It is now possible to add custom function to the Spel context in a simpler way. Try using `OfficeStamperConfiguration.addCustomFunction` methods.
* Added the `pro.verron.officestamper.utils` package to provide useful and common elements for core, preset, test and users.
* *Deprecated* the Paragraph wrapper now can provide a bit more advanced capability on inner P, don’t recommend calling `getP()` anymore.

### Dependencies

* Change java target from `17` to `21`
* Bump `org.springframework:spring-expression` from `6.1.12` to `6.1.14`
* Bump `org.springframework:spring-context` from `6.1.12` to `6.1.14`

### Continuous Integration

* Removed dependabot analysis
* Bump `org.apache.maven.plugins:maven-gpg-plugin` from `3.2.5` to `3.2.7`
* Bump `org.apache.maven.plugins:maven-javadoc-plugin` from `3.10.0` to `3.10.1`
* Bump `org.apache.maven.plugins:maven-surefire-plugin` from `3.5.0` to `3.5.1`
* Bump `org.apache.maven.reporting:maven-surefire-report-plugin` from `3.5.0` to `3.5.1`
* Bump `org.apache.maven.plugins:maven-site-plugin` from `3.20.0` to `3.21.0`
* Bump `org.apache.maven.plugins:maven-project-info-reports-plugin` from `3.7.0` to `3.8.0`
* Bump `org.apache.maven.plugins:maven-javadoc-plugin` from `3.10.0` to `3.10.1`
* Bump `org.apache.maven.reporting:maven-reporting-exec` from `2.0.0-M14` to `2.0.0`
* Bump `org.pitest:pitest-maven` from `1.16.3` to `1.17.0`
* Bump `org.asciidoctor:asciidoctor-maven-plugin` from `3.0.0` to `3.1.0`
* Bump `org.junit.jupiter:junit-jupiter` from `5.11.0` to `5.11.3`

### Refactorings

* CommentProcessor to now be setup with a P, run and placeholder with a ProcessorContext instead.
* Cleanup of malformed comments has moved to a PreProcessor instead of done on the fly during processing.
* Do a step toward treating Comment in the same way as Placeholder.
* *Deprecated* CommentProcessor to now be setup with a Paragraph that wraps and provides features around the docx4j P.
* *Deprecated* The `Image` preset object now deprecate `getMaxWidth` and `getImageBytes`.
* *BREAK* Removed DocxDocument.commentsPart method.
* *BREAK* DocxPart.streamParagraphs method now returns the Paragraph wrapper, instead of docx4j P.

### Tests

* Improved test names,
* Improved Word to string representation to better differentiate paragraph, run and cell styles.
* Remove exotic characters from string representation of Word documents.
* Section break and page break are more explicit in string representation of Word documents.

**Full Changelog**: https://github.com/verronpro/docx-stamper/compare/v2.5.0...v2.6.0

<hr /><em>This discussion was created from the release <a href='https://github.com/verronpro/docx-stamper/releases/tag/v2.6.0'>2.6.0</a>.</em>

**Comment by mpressmar** [comment]
Good day caring-coder,

thank you for your work, it's very much appreciated!

I noticed this version is not yet released to maven. Is that a usual delay, or did anything go wrong in the release process?

**Reply by mpressmar** [reply]
Thanks, it's available now!

**Reply by caring-coder** [reply]
I often have a delay with the maven release process. There is always a few pesky manual steps i didn't automate.
